### PR TITLE
value_expr: add charset value for string expr

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 

add charset info for string expr in restore function, such as `_utf8'string'`

### Check List 

Tests 

 - Unit test
Use the original test, no change
 - Integration test
test charset string in `parser/parser_test.go`

Code changes

 - Has exported function/method change
ValueExpr#Restore

### CODE REVIEW

**Comment by leoppro** [comment]
@kennytm @tiancaiamao  PTAL

**Comment by kennytm** [comment]
@leoppro Please fill in the Check List...

File: types/parser_driver/value_expr.go [hunk 1]
```diff
@@ -90,7 +90,12 @@ func (n *ValueExpr) Restore(ctx *format.RestoreCtx) error {
 		ctx.WritePlain(strconv.FormatFloat(n.GetFloat64(), 'e', -1, 32))
 	case types.KindFloat64:
 		ctx.WritePlain(strconv.FormatFloat(n.GetFloat64(), 'e', -1, 64))
-	case types.KindString, types.KindBytes:
+	case types.KindString:
+		if n.Type.Charset != "" && n.Type.Charset != "utf8mb4" {
+			ctx.WritePlainf("_%s", n.Type.Charset)
```
Comment (kennytm) [inline types/parser_driver/value_expr.go]:
```sql
mysql> select _UTF8'abc', _Utf8'abc', _utf8'abc';
+-----+-----+-----+
| abc | abc | abc |
+-----+-----+-----+
| abc | abc | abc |
+-----+-----+-----+
1 row in set, 3 warnings (0.01 sec)
```

Not sure if we should use `WritePlain` here...

Comment (leoppro) [reply types/parser_driver/value_expr.go]: i use WriteKeyWord now, PTAL

**Comment by kennytm** [comment]
LGTM.

cc @winkyao this involves charset.

File: types/parser_driver/value_expr.go [hunk 2]
```diff
@@ -90,7 +90,13 @@ func (n *ValueExpr) Restore(ctx *format.RestoreCtx) error {
 		ctx.WritePlain(strconv.FormatFloat(n.GetFloat64(), 'e', -1, 32))
 	case types.KindFloat64:
 		ctx.WritePlain(strconv.FormatFloat(n.GetFloat64(), 'e', -1, 64))
-	case types.KindString, types.KindBytes:
+	case types.KindString:
+		if n.Type.Charset != "" && n.Type.Charset != "utf8mb4" {
```
Comment (winkyao) [inline types/parser_driver/value_expr.go]: `if n.Type.Charset != "" && n.Type.Charset != mysql.DefaultCharset`

**Pull request merged**